tango_simlib/tango_sim_generator.py

```diff
@@ -214,14 +214,75 @@ def _reset_to_default_state(self):
+                        max_dim_y = 2
+                else:
+                    max_dim_x = 1
+                    max_dim_y = 2
+
+                val_type, val = INITIAL_CONSTANT_VALUE_TYPES[attr_data_type]
+                expected_key_vals = ['value', 'possiblevalues']
+                try:
+                    sim_quantity_meta_info['quantity_simulation_type']
+                except KeyError:
+                    if any(key_val in expected_key_vals for key_val in key_vals):
+                        try:
+                            adjustable_val = sim_quantity_meta_info['value']
+                        except KeyError:
+                            adjustable_val = sim_quantity_meta_info['possiblevalues']
+                        if attr_data_format == 'SCALAR':
+                            adjustable_val = val_type(adjustable_val)
+                        elif attr_data_format == 'SPECTRUM':
+                            adjustable_val = map(val_type, adjustable_val)
+                        else:
+                            adjustable_val = [[val_type(curr_val) for curr_val in sublist]
+                                for sublist in adjustable_val]
+                    else:
+                        if attr_data_format == 'SCALAR':
+                            adjustable_val = val
+                        elif attr_data_format == 'SPECTRUM':
+                            adjustable_val = [val] * max_dim_x
+                        else:
+                            adjustable_val = [[val] * max_dim_x for i in range(max_dim_y)]
+                    
+                    simulated_quantity.last_val = adjustable_val
+                else:
+                    if sim_quantity_meta_info['quantity_simulation_type'] == 'ConstantQuantity':
+                        try:
+                            initial_value = sim_quantity_meta_info['initial_value']
+                        except KeyError:
+                            initial_value = None
+                        adjustable_val = (initial_value if initial_value not in [None, ""]
+                                else val)
+                        if val_type is None:
+                            adjustable_val = None
+                        else:
+                            adjustable_val = val_type(adjustable_val)
+                        simulated_quantity.last_val = adjustable_val
+                    else:
+                        simulated_quantity.mean = float(sim_quantity_meta_info['min_bound'])
+                        simulated_quantity.max_bound = float(sim_quantity_meta_info['max_bound'])
+                        simulated_quantity.max_slew_rate = float(sim_quantity_meta_info['max_slew_rate'])
+                        simulated_quantity.mean = float(sim_quantity_meta_info['mean'])
+                        simulated_quantity.std_dev = float(sim_quantity_meta_info['std_dev'])
+                        simulated_quantity.last_val = float(sim_quantity_meta_info['mean'])
```

Comment:
Previously we didn't have to update all of these fields explicitly.  That was because we were iterating through the `adjustable_attributes`.  For the `GaussianSlewLimited` type, these are included:
https://github.com/ska-sa/tango-simlib/blob/43e5567b0c20ec7da9599d51f5411b6b1b6c2bb7/tango_simlib/quantities.py#L124-L125

The solution here is now less general than it used to be, and ignores the `adjustable_attributes` which makes it brittle - e.g. if we add a new one.   The only special case, where the name one the left differs to the name on the right is `last_val` which is set to `mean`.  Suggest we keep something like the old loop that takes care of the attributes generically, around line 267.  Then after that is done, check the type of quantity and set the `last_val` to whatever makes sense for each type.  If the quantity type is unknown, then we could raise a `NotImplementedError`.

Reply:
thanks for pointing this out